python/kwiver/sprokit/tests/sprokit/adapters/test-embedded_pipeline.py

```diff
@@ -0,0 +1,154 @@
+#    to endorse or promote products derived from this software without specific
+#    prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+
+def test_import(pipeline_dir):
+    try:
+        import kwiver.sprokit.adapters.embedded_pipeline
+    except:
+        test_error("Failed to import the embedded pipeline module")
+
+
+def test_create(pipeline_dir):
+    from kwiver.sprokit.adapters import embedded_pipeline
+
+    embedded_pipeline.EmbeddedPipeline()
+
+
+def test_api_calls(pipeline_dir):
+    from kwiver.vital.config import config
+    from kwiver.vital.modules import modules
+    from kwiver.sprokit.adapters import embedded_pipeline
+    from kwiver.sprokit.pipeline import process
+    from kwiver.sprokit.adapters import adapter_data_set
+
+    #######
+    # Run a very basic pipeline.
+    # Pipe file and tests below are
+    # based on embedded_pipeline tests
+    # in basic_test.cxx in adapters/tests dir
+
+    pipeline_fname = "simple_embedded_pipeline.pipe"
+    path_to_pipe_file = os.path.join(pipeline_dir, pipeline_fname)
+
+    ep = embedded_pipeline.EmbeddedPipeline()
+    ep.build_pipeline(path_to_pipe_file)
+
+    # Check the input_ports
+    input_list = ep.input_port_names()
+    if len(input_list) != 3:
+        test_error(
+            "input_port_list() returned list of length {}, expected {}".format(
+                len(input_list), 3
+            )
+        )
+    for port in input_list:
+        print("    " + port)
+
+    # Check the output ports
+    output_list = ep.output_port_names()
+    if len(output_list) != 3:
+        test_error(
+            "output_port_list() returned list of length {}, expected {}".format(
+                len(output_list), 3
+            )
+        )
+    for port in output_list:
+        print("    " + port)
+
+    # Test that we can't call wait() yet
+    try:
+        ep.wait()
+    except RuntimeError:
+        pass
+    else:
+        test_error("Calling wait() before start() should throw an error")
+
+    ep.start()
+
+    # Now send some data
+    for i in range(10):
+        ds = adapter_data_set.AdapterDataSet.create()
+        val = i
+
+        for port in input_list:
+            ds.add_int(port, val)
+            val += 1
+
```

Comment:
Logic can be simplified as such: 

```python
for val, port in enumerate(input_list, start=i):
     ds.add_int(port, val)
```

No need for `val =i` and `val += 1`

Reply:
Much better. Resolved in e7dbafbca27d1c95722c142fe333969ce1200319